scipy/signal/spectral.py

```diff
@@ -511,11 +511,481 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+        Otherwise, ``nperseg=Zxx.shape[freq_axis]``.
+    noverlap : int, optional
+        Number of points to overlap between segments. If ``None``, half of the
+        segment length. Defaults to ``None``. When specified, the COLA constraint
+        must be met (see Notes below), and should match the parameter used to
+        generate the STFT.
+    nfft : int, optional
+        Number of FFT points corresponding to each STFT segment. This parameter
+        must be specified if the STFT was padded via ``nfft > nperseg``. If
+        ``None``, the default values are the same as for ``nperseg``, detailed
+        above, with one exception: if ``input_onesided`` is ``True`` and
+        ``nperseg==2*Zxx.shape[freq_axis] - 1``, ``nfft`` also takes on that
+        value. This case allows the proper inversion of an odd-length unpadded
+        STFT using ``nfft=None``.
+    input_onesided : bool, optional
+        If ``True``, interpret the input array as one-sided FFTs, such as is
+        returned by `numpy.fft.rfft`. If ``False``, interpret the input as a a
+        two-sided FFT.
+    centered : bool, optional
+        Specifies whether the input signal was padded via odd extension, by
+        using ``centered=True`` in `stft`. Defaults to ``True``.
+    time_axis : int, optional
+        Where the time segments of the STFT is located; the default is the
+        last axis (i.e. ``axis=-1``).
+    freq_axis : int, optional
+        Where the frequency axis of the STFT is located; the default is the
+        penultimate axis (i.e. ``axis=-2``).
+
+    Returns
+    -------
+    t : ndarray
+        Array of output data times.
+    x : ndarray
+        iSTFT of ``Zxx``.
+
+    See Also
+    --------
+    stft: Short Time Fourier Transform
+    check_COLA: Check whether the Constant OverLap Add (COLA) constraint is met
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in `istft`,
+    the signal windowing must obey the constraint of "Constant OverLap Add"
+    (COLA). This ensures that every point in the input data is equally
+    weighted, thereby avoiding aliasing and allowing full reconstruction.
+    Whether a choice of ``window``, ``nperseg``, and ``noverlap`` satisfy this
+    constraint can be tested with `check_COLA`, by using ``nperseg =
+    Zxx.shape[freq_axis]``.
+
+    .. versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Oppenheim, Alan V., Ronald W. Schafer, John R. Buck
+           "Discrete-Time Signal Processing", Prentice Hall, 1999.
+    .. [2] Daniel W. Griffin, Jae S. Limdt "Signal Estimation from Modified
+           Short Fourier Transform", IEEE 1984, 10.1109/TASSP.1984.1164317
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> from scipy.fftpack import fftshift
+    >>> import matplotlib.pyplot as plt
+
+    Generate a test signal, a 2 Vrms sine wave at 100Hz corrupted by 0.001
+    V**2/Hz of white noise sampled at 1024 Hz.
```

Comment:
My guess is that most data in the wild are not sampled at a nice multiple of 2. Does the example still work (hopefully yes) if you use a 1000 Hz sample rate?